esphome/components/bme68x_bsec2_i2c/__init__.py

```diff
@@ -0,0 +1,211 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+bme68x_bsec2_i2c_ns = cg.esphome_ns.namespace("bme68x_bsec2_i2c")
+
+MODEL_OPTIONS = ["bme680", "bme688"]
+
+AlgorithmOutput = bme68x_bsec2_i2c_ns.enum("AlgorithmOutput")
+ALGORITHM_OUTPUT_OPTIONS = {
+    "classification": AlgorithmOutput.ALGORITHM_OUTPUT_CLASSIFICATION,
+    "regression": AlgorithmOutput.ALGORITHM_OUTPUT_REGRESSION,
+}
+
+OperatingAge = bme68x_bsec2_i2c_ns.enum("OperatingAge")
+OPERATING_AGE_OPTIONS = {
+    "4d": OperatingAge.OPERATING_AGE_4D,
+    "28d": OperatingAge.OPERATING_AGE_28D,
+}
+
+SampleRate = bme68x_bsec2_i2c_ns.enum("SampleRate")
+SAMPLE_RATE_OPTIONS = {
+    "LP": SampleRate.SAMPLE_RATE_LP,
+    "ULP": SampleRate.SAMPLE_RATE_ULP,
+}
+
+Voltage = bme68x_bsec2_i2c_ns.enum("Voltage")
+VOLTAGE_OPTIONS = {
+    "1.8V": Voltage.VOLTAGE_1_8V,
+    "3.3V": Voltage.VOLTAGE_3_3V,
+}
+
+BME68xBSEC2I2CComponent = bme68x_bsec2_i2c_ns.class_(
+    "BME68xBSEC2I2CComponent", cg.Component, i2c.I2CDevice
+)
+
+ALGORITHM_OUTPUT_FILE_NAME = {
+    "classification": "sel",
+    "regression": "reg",
+}
+
+SAMPLE_RATE_FILE_NAME = {
+    "LP": "3s",
+    "ULP": "300s",
+}
+
+VOLTAGE_FILE_NAME = {
+    "1.8V": "18v",
+    "3.3V": "33v",
+}
+
+
+def validate_bme68x(config):
+    if CONF_ALGORITHM_OUTPUT not in config:
+        return config
+
+    if config[CONF_MODEL] != "bme688":
+        raise cv.Invalid(f"{CONF_ALGORITHM_OUTPUT} is only valid for BME688")
+
+    if config[CONF_ALGORITHM_OUTPUT] == "regression" and (
+        config[CONF_OPERATING_AGE] != "4d"
+        or config[CONF_SAMPLE_RATE] != "ULP"
+        or config[CONF_SUPPLY_VOLTAGE] != "1.8V"
+    ):
+        raise cv.Invalid(
+            f" To use '{CONF_ALGORITHM_OUTPUT}: regression', {CONF_OPERATING_AGE} must be '4d', {CONF_SAMPLE_RATE} must be 'ULP' and {CONF_SUPPLY_VOLTAGE} must be '1.8V'"
+        )
+    return config
+
+
+CONFIG_SCHEMA = cv.All(
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(BME68xBSEC2I2CComponent),
+            cv.GenerateID(CONF_RAW_DATA_ID): cv.declare_id(cg.uint8),
+            cv.Required(CONF_MODEL): cv.one_of(*MODEL_OPTIONS, lower=True),
+            cv.Optional(CONF_ALGORITHM_OUTPUT): cv.enum(
+                ALGORITHM_OUTPUT_OPTIONS, lower=True
+            ),
+            cv.Optional(CONF_OPERATING_AGE, default="28d"): cv.enum(
+                OPERATING_AGE_OPTIONS, lower=True
+            ),
+            cv.Optional(CONF_SAMPLE_RATE, default="LP"): cv.enum(
+                SAMPLE_RATE_OPTIONS, upper=True
+            ),
+            cv.Optional(CONF_SUPPLY_VOLTAGE, default="3.3V"): cv.enum(
+                VOLTAGE_OPTIONS, upper=True
+            ),
+            cv.Optional(CONF_TEMPERATURE_OFFSET, default=0): cv.temperature,
+            cv.Optional(
+                CONF_STATE_SAVE_INTERVAL, default="6hours"
+            ): cv.positive_time_period_minutes,
+        },
+        cv.only_with_arduino,
+    ).extend(i2c.i2c_device_schema(0x76)),
+    validate_bme68x,
```

Comment:
I would put these outside together
```suggestion
    ).extend(i2c.i2c_device_schema(0x76)),
    cv.only_with_arduino,
    validate_bme68x,
```